pkg/client/app.go

```diff
@@ -106,7 +106,8 @@ func ToAppUpdate(ctx context.Context, c Client, name string, opts *AppUpdateOpti
 		return app, nil
 	}
 
-	if opts.Image != "" {
+	// only change app.Spec.Image if the user specified a different image
+	if opts.Image != "" && opts.Image != app.Spec.Image {
```

Comment:
This if statement contains the logic that finds a new image if the user specified an image for the update. In this case, we only want to do this if an image is defined (`opts.Image != ""`) and the image defined is not the same as the one currently in the app spec (`opts.Image != app.Spec.Image`). Only then do we go looking for the new image and change the app spec based on it.